lib/mongoose_push/service/apns/certificate.ex

```diff
@@ -0,0 +1,46 @@
+defmodule MongoosePush.Service.APNS.Certificate do
+  @moduledoc false
+
+  require Record
+  Record.defrecord :otp_cert,
+    Record.extract(:OTPCertificate, from_lib: "public_key/include/public_key.hrl")
+  Record.defrecord :tbs_cert,
+    Record.extract(:TBSCertificate, from_lib: "public_key/include/public_key.hrl")
+  Record.defrecord :cert_ext,
+    Record.extract(:Extension,      from_lib: "public_key/include/public_key.hrl")
+
+  # From: https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CommunicatingwithAPNs.html
+  @apns_topic_extn_id {1, 2, 840, 113635, 100, 6, 3, 6}
+
+  def extract_topics!(cert_file) do
+    maybe_extension =
+      cert_file
+      |> File.read!()
+      |> :public_key.pem_decode()
+      |> List.keyfind(:Certificate, 0)
+      |> get_cert_extension!(@apns_topic_extn_id)
+
+    if maybe_extension == nil do
+        throw :no_extension
```

Comment:
This function is meant to `try` load APNS topic. If it fails, for *whatever* reason, `maybe_setup_default_apns_topic` should fallback to other means of getting the topic. Again - the reason of the failure could be less important. The only thing that the reason is helpful of is that we should log the failures in order to inform user that we have to fallback to other means of getting topic and possibly why. Therefore, this `if` is here only to change log message to be more readable in this one case. 

Elixir's `raise` is meant for special error conditions, that maybe need special handling. Here, theres no different route of handling this exception no matter if it looks like that, or is just a badmatch, which would be the case if the `if` wasn't there.